decompile.py

```diff
@@ -40,33 +41,30 @@ def from_raw(cls, data, frontend: Frontend = BinaryninjaFrontend) -> Decompiler:
         """Create a decompiler instance from existing frontend instance (e.g. a binaryninja view)."""
         return cls(frontend.from_raw(data))
 
-    def decompile(self, function: str, task_options: Optional[Options] = None) -> DecompilerTask:
-        """Decompile the target function."""
-        # Sanity check to ensure task_options is populated
+    def decompile(self, function_ids: Collection[object] | None = None, task_options: Options | None = None) -> Result:
```

Comment:
I did this because both function where almost the same code expect that one used a loop to decompile multiple functions. So in other ones the functionally of decompiling one function is already covered by the decompiling of multiple functions. I think that having the code duplicated makes it more prone to errors.
Instead if we still want both methods, we could introduce a helper method like I've shown in https://github.com/fkie-cad/dewolf/pull/391#discussion_r1549831763, which simply delegates to the decompile function.